packages/shared/src/open-feature.ts

```diff
@@ -204,27 +264,27 @@ export abstract class OpenFeatureCommonAPI<P extends CommonProvider = CommonProv
 
     const emitters = this.getAssociatedEventEmitters(domain);
 
-    // warn of improper implementations
-    if (typeof provider.initialize === 'function' && provider.status === undefined) {
-      const activeLogger = this._logger || console;
-      activeLogger.warn(
-        `Provider ${providerName} implements 'initialize' but not 'status'. Please implement 'status'.`,
-      );
-    }
-
     let initializationPromise: Promise<void> | void = undefined;
+    const wrappedProvider = new ProviderWrapper<P, AnyProviderStatus>(provider, this._statusEnumType.NOT_READY, this._statusEnumType);
 
-    if (provider?.status === ProviderStatus.NOT_READY && typeof provider.initialize === 'function') {
+    if (typeof provider.initialize === 'function') {
```

Comment:
The provider might already be initialized here, if it has been assigned to another domain right?
Before we checked it and now we are just reinitializing. We are not specifying that the `initialize` function should be idempotent and reentrant or so.
I think this might break with other implementations and think that we should maybe look in `allProviders` if this exact provider instance is already assigned and in which state it is.

Reply:
Yes I think you're right about this. I will add such a check.

Reply:
I created a new `allProviders` getter since we were already using logic that needed that elsewhere, and used it here for this check: https://github.com/open-feature/js-sdk/pull/795/commits/8cbd24da28cc79f67abdea3fb4ec61e284380028